conjure-java-jersey-server/src/main/java/com/palantir/conjure/java/server/jersey/ConjureJerseyFeature.java

```diff
@@ -86,4 +87,81 @@ public boolean configure(FeatureContext context) {
 
         return true;
     }
+
+    @CheckReturnValue
+    static Builder builder() {
+        return new Builder();
+    }
+
+    @CheckReturnValue
+    public static final class Builder {
+        private ExceptionListener exceptionListener = NoOpListener.INSTANCE;
+
+        /**
+         * Every throwable handled by the {@code ConjureJerseyFeature} is first passed to this {@code
+         * exceptionListener}. This is a good opportunity to record metrics about the different types of exceptions.
+         */
+        Builder exceptionListener(ExceptionListener value) {
+            this.exceptionListener = value;
+            return this;
+        }
+
+        Feature build() {
+            ExceptionListener listener = Preconditions.checkNotNull(exceptionListener, "exceptionListener");
+            return new Feature() {
+                @Override
+                public boolean configure(FeatureContext context) {
+                    return ConjureJerseyFeature.configure(context, listener);
+                }
+
+                @Override
+                public String toString() {
+                    return "ConjureJerseyFeature{}";
+                }
+            };
+        }
+    }
+
+    /** Implementors of this interface can. */
+    public interface ExceptionListener {
+
+        void onException(Throwable throwable);
+
+        /** Invoked in a finally block, so use this to unset any MDC values. */
+        void afterResponseBuilt();
+
+        default <T extends Throwable> ExceptionMapper<T> augment(ExceptionMapper<T> delegate) {
+            return new ExceptionMapper<T>() {
+                @Override
+                public Response toResponse(T exception) {
+                    try {
+                        onException(exception);
+                        return delegate.toResponse(exception);
+                    } finally {
+                        afterResponseBuilt();
```

Comment:
Not sure the abstract methods really make sense here, I think this interface only needs to have a single method which consumes a Throwable.

Reply:
That's what I had originally, are you thinking we just do MDC.clear() afterwards?

Reply:
Sorry, wasn't quite a complete thought.

Since this interface deals directly with ExceptionMapper, I was thinking it might as well operate as an ExceptionMapper wrapper so implementations can use try/finally. Not a strong preference though, and it's probably fine as is.

Reply:
Ugh, seems like this 100% does not work with jersey (I think due to type erasure). 

```
Caused by: javax.ws.rs.ProcessingException: Could not find exception type for given ExceptionMapper class: class com.palantir.conjure.java.server.jersey.ConjureJerseyFeature$ExceptionListener$1.
	at org.glassfish.jersey.internal.ExceptionMapperFactory.getType(ExceptionMapperFactory.java:260)
	at org.glassfish.jersey.internal.ExceptionMapperFactory.getExceptionType(ExceptionMapperFactory.java:230)
	at org.glassfish.jersey.internal.ExceptionMapperFactory.lambda$createLazyExceptionMappers$1(ExceptionMapperFactory.java:204)
	at org.glassfish.jersey.internal.util.collection.Values$LazyValueImpl.get(Values.java:317)
	at org.glassfish.jersey.internal.ExceptionMapperFactory.find(ExceptionMapperFactory.java:115)
	at org.glassfish.jersey.internal.ExceptionMapperFactory.findMapping(ExceptionMapperFactory.java:102)
	at org.glassfish.jersey.server.ServerRuntime$Responder.mapException(ServerRuntime.java:523)
	at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:403)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:261)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
```

Gonna plumb these in manually and see if that does the trick.

Reply:
Ya, that makes sense. I think we need to provide the listener to each exception mapper.

Reply:
Can reproduce the issue by removing the `ExceptionListener.augment` override from `NoOpListener`.